catkin_ws/src/sensors/src/hydrophones_bearing.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import rospy
+
+import numpy as np
+from std_msgs.msg import Float64MultiArray
+
+
+def cb_hydrophones(msg):
+    dt_h1 = msg.data[0]
+    dt_h2 = msg.data[1]
+
+    measurements = calculate_time_measurements(dt_h1, dt_h2) 
+    bearing_vector = solve_bearing_vector(measurements[0], measurements[1])
+    bearing_vector_msg = Float64MultiArray()
+    bearing_vector_msg.data = bearing_vector
+    pub_pinger_direction.publish(bearing_vector_msg)
+
+#With 4 hydrophones
+def solve_bearing_vector(H_matrix, measurements, num):
+    H_matrix = np.array(H_matrix)
+    measurements = np.array(measurements)
+
+    # Check if invertible
+    if np.linalg.matrix_rank(H_matrix) < num:
+        raise ValueError("Not invertible H_matrix")
+
+    bearing_vector = np.linalg.inv(H_matrix) @ measurements
+
+    return bearing_vector
+
+def calculate_time_measurements(t0,t1,t2,t3):
+    '''
+    tn = time of arrival at hydrophone n
+    c = speed of sound in medium
+    dn = c(t0-tn)
+    Take t0 as the reference time (time when sound wave hits hydrophone 0)
+    '''
+    d1 = c*(t0-t1)
+    d2 = c*(t0-t2)
+    d3 = c*(t0-t3)
+    return np.array([d1,d2,d3])
+
```

Comment:
this file seems bloated, do we need all these functions?

Reply:
it looks fine to me I don't think we should nitpick how they implement things if it looks good and works

Reply:
its mainly because there's two copies of two functions, if its necessary thats fine but it looks like one of them is never used

Reply:
i see what you mean maybe we should name them differently

Reply:
also I think this pr is still a work in progress so we can hold our comments until its ready

Reply:
ah ok my bad I got an email about it and saw my review requested, I'll un-request the reviews

Reply:
This is not finished guys